sympy/series/limits.py

```diff
@@ -109,6 +109,16 @@ def __new__(cls, e, z, z0, dir="+"):
         obj._args = (e, z, z0, dir)
         return obj
 
+
+    @property
+    def free_symbols(self):
+        e = self.args[0]
+        isyms = (e - self.args[1]).free_symbols
+        isyms.discard(self.args[1])
```

Comment:
I think that a limit like ``Limit(x + y, sin(x), 2)`` is meaningless. Therefore using its free_symbols will lead to wrong results in any case.

But your code in the present form is very sensitive to the structure of the expression.
```
>>> Limit(x**2 + y, x**2, a).free_symbols
set([a, y])
>>> Limit(-x**2 + y, x**2, a).free_symbols
set([x, a, y])
```